main.cpp

```diff
@@ -0,0 +1,79 @@
+//
+//  main.cpp
+//  CodeU_ass1
+//
+//  Created by Elizaveta on 18/05/2017.
+//  Copyright © 2017 Elizaveta. All rights reserved.
+//
+
+#include <iostream>
+#include <string>
+#include <algorithm>
+
+using namespace std;
+
+string is_permutation(string str1, string str2) {
+    transform(str1.begin(), str1.end(), str1.begin(), ::tolower);
+    transform(str2.begin(), str2.end(), str2.begin(), ::tolower);
+    
+    sort(str1.begin(), str1.end());
+    sort(str2.begin(), str2.end());
+    
+    return str1 == str2 ? "True" : "False";
+}
+
+struct Node {
+    Node* head;
+    Node* next;
+    int value;
+};
+
+void find_k_to_end(struct Node* head, int k, int &result_value) {
```

Comment:
I think is better to return a Node * than the value, that gives the caller more flexibility.
Also, why use a output parameter instead of the return value?
A generic comment on this function, the statement specifies that the last node should be returned when k=0, the previous when k=1, etc.